pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -555,31 +574,35 @@ func applyHTTPRouteDestination(
 		}
 	}
 
+	var hostnames []host.Name
 	if len(in.Route) == 1 {
-		processDestination(in.Route[0], serviceRegistry, listenerPort, hashByDestination, out, action)
+		hostnames = append(hostnames, processDestination(in.Route[0], serviceRegistry, listenerPort, hashByDestination, out, action))
 	} else {
 		weighted := make([]*route.WeightedCluster_ClusterWeight, 0)
 		for _, dst := range in.Route {
 			if dst.Weight == 0 {
 				// Ignore 0 weighted clusters if there are other clusters in the route.
 				continue
 			}
-			weighted = append(weighted, processWeightedDestination(dst, serviceRegistry, listenerPort, hashByDestination, action))
+			destinationweight, hostname := processWeightedDestination(dst, serviceRegistry, listenerPort, hashByDestination, action)
+			weighted = append(weighted, destinationweight)
+			hostnames = append(hostnames, hostname)
 		}
 		action.ClusterSpecifier = &route.RouteAction_WeightedClusters{
 			WeightedClusters: &route.WeightedCluster{
 				Clusters: weighted,
 			},
 		}
 	}
+	return hostnames
 }
 
 func processDestination(dst *networking.HTTPRouteDestination, serviceRegistry map[host.Name]*model.Service,
```

Comment:
Maybe a small comment - what does hostname contain ? ( and what does the method do - the name is pretty vague ). I would assume the result is the list of all 'hostnames' that are used in any of the destinations ?